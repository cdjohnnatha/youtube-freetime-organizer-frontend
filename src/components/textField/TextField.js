
/* eslint-disable react/react-in-jsx-scope */
import React from 'react';
import FormControl from '@material-ui/core/FormControl';
import FormHelperText from '@material-ui/core/FormHelperText';
import CustomTextField from '@material-ui/core/TextField';
import { string } from 'prop-types';

const TextField = ({
  label,
  format,
  error,
  field,
  form,
  ...otherProps
}) => {
  let extraProps = {};

  if (form) {
    const { touched, errors } = form;
    error = touched[field.name] && errors[field.name] ? errors[field.name] : '';
  }

  if (format) {
    const currentInputProps = otherProps.InputProps ? otherProps.InputProps : {};
    extraProps.InputProps = {
      ...currentInputProps,
      inputProps: { format }
    };
  }

  return (
    <FormControl fullWidth error={error ? true : false}>
      <CustomTextField
        {...otherProps}
        {...field}
        {...extraProps}
        label={label}
        error={error ? true : false}
      />
      {error && <FormHelperText>{error}</FormHelperText>}
    </FormControl>
  );
};

TextField.defaultProps = {
  // validationRegex: alphaNumericCharactersAllowed
};

TextField.propTypes = {
  label: string
};

export default TextField;