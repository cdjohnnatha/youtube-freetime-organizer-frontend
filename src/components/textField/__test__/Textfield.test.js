
import React from 'react';
import { create } from 'react-test-renderer';
import TextField from '../TextField';


describe('Test TextField component', () => {
  test('TextField snapshot renders', () => {
    const component = create(<TextField />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('TextField component', () => {
    const component = create(
      <TextField />
    ).root;
    const emptyContent = component.findByType('input');
    expect(emptyContent.type).toBe('input');
  });
});