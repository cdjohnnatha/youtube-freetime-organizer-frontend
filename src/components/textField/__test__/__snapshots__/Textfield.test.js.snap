// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TextField component TextField snapshot renders 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth"
>
  <div
    className="MuiFormControl-root MuiTextField-root"
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
    </div>
  </div>
</div>
`;
